in_toto/models/mock_link.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""
+<Program Name>
+  mock_link.py
+
+<Author>
+  Shikher Verma <root@shikherverma.com>
+
+<Started>
+  June, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides a class for mock link metadata which is information gathered when a
+  step of the supply chain is run as mock.
+"""
+
+import attr
+import json
+from . import link
+import securesystemslib.formats
+
+MOCK_FILENAME_FORMAT = "{step_name}.link-mock"
+
+@attr.s(repr=False, init=False)
+class MockLink(link.Link):
+  """
+  A mock link is the metadata representation of a supply chain step mocked
+  by a functionary.
+
+  Mock links are recorded and stored to a file when a functionary
+  wraps a command with in_toto-run.
+
+  Mock links also contain materials and products which are hashes of
+  the file before the command was executed and after the command was executed.
+
+  <Attributes>
+    name:
+        a unique name used to identify the related step in the layout
+    materials and products:
+        a dictionary in the format of
+          { <relative file path> : {
+            {<hash algorithm> : <hash of the file>}
+          },... }
+
+    byproducts:
+        a dictionary in the format of
+          {
+            "stdout": <standard output of the executed command>
+            "stderr": <standard error of the executed command>
+          }
+
+    command:
+        the command that was wrapped by in_toto-run
+
+    return_value:
+        the return value of the executed command
+
+    working_directory:
+        the path of the directory where the command was executed
+   """
+  _type = attr.ib()
+  name = attr.ib()
+  materials = attr.ib()
+  products = attr.ib()
+  byproducts = attr.ib()
+  command = attr.ib()
+  return_value = attr.ib()
+  working_directory = attr.ib()
+
+  def __init__(self, **kwargs):
+    super(MockLink, self).__init__(**kwargs)
+
+    self._type = "MockLink"
+    self.name = kwargs.get("name")
+    self.materials = kwargs.get("materials", {})
+    self.products = kwargs.get("products", {})
+    self.byproducts = kwargs.get("byproducts", {})
+    self.command = kwargs.get("command", [])
+    self.return_value = kwargs.get("return_value", None)
```

Comment:
Actually, you can let the inherited attributes be assigned in the parent constructor.